learn.html

```diff
@@ -0,0 +1,447 @@
+							<p class="c_snip">
+								release=$(awk '{x=$7; split(x, a, "."); print a[1];}' /etc/redhat-release)<br>
+								case $release in<br>
+								&nbsp;&nbsp;6)<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;pkg=cpupowerutils<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;;;<br>
+								&nbsp;&nbsp;7)<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;pkg=kernel-tools<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;;;<br>
+								&nbsp;&nbsp;*)<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# better be installed already<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; ;;<br>
+								esac<br>
+
+								case "$mode" in<br>
+								&nbsp;&nbsp;install)<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;if [ ! -e $tool_bin ]; then<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;yum install $pkg<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;echo $script_name is installed<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;else<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;echo $script_name is installed<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;fi<br>
+								&nbsp;&nbsp;start)<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;mkdir -p $tool_output_dir<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;echo "$tool_cmd" >$tool_cmd_file<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;debug_log "$script_name: running $tool_cmd"<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;$tool_cmd >>"$tool_output_file" & echo $! >$tool_pid_file<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;wait<br>
+								&nbsp;&nbsp;&nbsp;&nbsp;;;<br>
+								&nbsp;&nbsp;stop)<br>
+								&nbsp;&nbsp; &nbsp;&nbsp;pid=`cat "$tool_pid_file"`<br>
+								&nbsp;&nbsp; &nbsp;&nbsp;debug_log "stopping $script_name"<br>
+								&nbsp;&nbsp; &nbsp;&nbsp;kill $pid && /bin/rm "$tool_pid_file"<br>
+								&nbsp;&nbsp; &nbsp;&nbsp;;;<br>
+								&nbsp;&nbsp;postprocess)<br>
+								&nbsp;&nbsp; &nbsp;&nbsp;debug_log "postprocessing $script_name"<br>
+								&nbsp;&nbsp; &nbsp;&nbsp;$script_path/postprocess/$script_name-postprocess $tool_output_dir<br>
+								&nbsp;&nbsp; &nbsp;&nbsp;;;<br>
+								esac
+							</p>
+							<p class="info_text" style="text-align: justify">Finally, there is the post-processing tool: the simplest thing to do is nothing. That's currently the case for the <a href="https://github.com/distributed-system-analysis/pbench/blob/master/agent/tool-scripts/kvm-spinlock">turbostat</a> post-processing tool, but ideally it should produce a JSON file with the data points and an HTML file that uses the nv3 library to plot the data graphically in a browser. See the <a href="https://github.com/distributed-system-analysis/pbench/tree/master/agent/tool-scripts/postprocess">postprocess</a> directory for examples, e.g. <a href="https://github.com/distributed-system-analysis/pbench/blob/master/agent/tool-scripts/postprocess/iostat-postprocess">iostat postprocessing tool.</a></p>
```

Comment:
the href says "kvm-spinlock" but the text says "turbostat" ... I'm assuming this probably wasn't deliberate?

Reply:
Yeah. I did a mistake here. But the link for "[turbostat](https://github.com/distributed-system-analysis/pbench/blob/master/agent/tool-scripts/postprocess/turbostat-postprocess)" given in documentation isn't working as well.